processor/metricextractprocessor/processor.go

```diff
@@ -0,0 +1,196 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricextractprocessor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/observiq/observiq-otel-collector/internal/expr"
+	"github.com/observiq/observiq-otel-collector/receiver/routereceiver"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+// processor is a processor that extracts metrics from logs.
+type processor struct {
```

Comment:
nit: Can we change the name to `metricExtractProcessor` or something more specific. The latest version of otel introduced the following packages `receiver`, `exporter`, `extension`. Which caused collision in some of our code where we had those same names for structs. I have a feeling at some point there will be a `processor` package as well.